pkg/pub_worker/lib/src/bin/dartdoc_wrapper.dart

```diff
@@ -85,14 +85,33 @@ Future<void> postProcessDartdoc({
     _log.info('Finished .tar.gz archive');
   });
 
-  await Future.wait([
-    processingFuture,
-    archiveFuture,
-  ]);
+  try {
+    await Future.wait([
+      processingFuture,
+      archiveFuture,
+    ]);
 
-  // Move from temporary output directory to final one, ensuring that
-  // documentation files won't be present unless all files have been processed.
-  // This helps if there is a timeout along the way.
-  await Directory(tmpOutDir).rename(p.join(outputFolder, 'doc'));
-  await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+    // Move from temporary output directory to final one, ensuring that
+    // documentation files won't be present unless all files have been processed.
+    // This helps if there is a timeout along the way.
+    await tmpOutDir.rename(p.join(outputFolder, 'doc'));
+    await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+  } catch (e, st) {
+    // We are likely out of free disk space.
+    // Cleanup the content as we won't use it.
+    await tmpOutDir.deleteIgnoringErrors(recursive: true);
+    await tmpTar.deleteIgnoringErrors();
+
+    _log.shout('Failed to run dartdoc post-processing.', e, st);
+  }
```

Comment:
So now now it'll just complete successfully without any content.

Are we sure that's the behavior we want?

The log messages here go to stackdriver. Not the package task-log.

Which is IMO also the right thing.

But successfully completing the task is probably not desirable. Or at-least if we do, we should write in the task-log that something failed.

Reply:
To be fair, I'm not sure we fully understand what it is we want to do. We should perhaps chat a bit more about this.